azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageVersionDaoImpl.java

```diff
@@ -103,6 +103,10 @@
           + "iv.created_by, iv.modified_on, iv.modified_by from image_versions iv, image_types it where "
           + "it.id = iv.type_id and lower(it.name) = ? and iv.id = ?";
 
+  private static final String SELECT_INVALID_IMAGE_VERSION_QUERY = "select not exists (select "
+      + "iv.version from image_versions iv inner join image_types it on it.id = iv.type_id "
+      + "where lower(it.name) = ? and iv.version = ? and iv.state in (?, ?)) is_invalid";
```

Comment:
Yes, we need to put appropriate index the required column. This can be taken up separately as part of db change. Actually, we need to come up with the index for image management tables wherever necessary. So we can create a new task and take this up separately.

Reply:
Sure. I have created this task and added it as a part of sprint.